pkg/generic/thrift/write.go

```diff
@@ -323,14 +321,93 @@ func writeBool(ctx context.Context, val interface{}, out thrift.TProtocol, t *de
 }
 
 func writeInt8(ctx context.Context, val interface{}, out thrift.TProtocol, t *descriptor.TypeDescriptor, opt *writerOption) error {
+	var i int8
 	switch val := val.(type) {
 	case int8:
-		return out.WriteByte(val)
+		i = val
 	case uint8:
-		return out.WriteByte(int8(val))
+		i = int8(val)
 	default:
 		return fmt.Errorf("unsupported type: %T", val)
 	}
+	// compatible with lossless conversion
+	switch t.Type {
+	case thrift.I08:
+		return out.WriteByte(i)
+	case thrift.I16:
+		return out.WriteI16(int16(i))
```

Comment:
It seems not right. The receiving end will get an int16, and may fail to decode. We should adhere to the thrift binary protocol here.

Reply:
Sorry my fault. It should be right.